tooling/scripts/sanity-checks.groovy

```diff
@@ -68,10 +68,11 @@ int compareVersion(String a, String b) {
     List verB = b.tokenize('.')
 
     def commonIndices = Math.min(verA.size(), verB.size())
+    def versionRegex = ~/\-.*/
 
     for (int i = 0; i < commonIndices; ++i) {
        def numA = verA[i].toInteger()
-       def numB = verB[i].toInteger()
+       def numB = (verB[i] - versionRegex).toInteger()
```

Comment:
I am bit lost what is this doing. The minus operator removes the matching part from the string, right?
I this way we are effectively disregard the `-RC` qualifiers, right? 
That's definitely OK for this current versions check, but I wonder whether it would not be safer for future copy-and-paste visitors, to move the suffix stripping before the compareVersion() method. After all we strip `-SNAPSHOT` from project.version before passing it to compareVersion() too.

Reply:
Yeah, it intends to strip `-RC`. So Maybe it could move before `compareVersion()` like
```java
final String currentVersion = project.version - '~/\-.*/'
```

Reply:
Yeah, I think that would be safer for any future reuse of `compareVersion()`

Reply:
OK, I fix it.